astropy/cosmology/flrw.py

```diff
@@ -1485,6 +1454,99 @@ def arcsec_per_kpc_proper(self, z):
         return u.arcsec / (self.angular_diameter_distance(z).to(u.kpc) *
                            arcsec_in_radians)
 
+    # -----------------------------------------------------
+
+    def __equiv__(self, other):
+        """FLRW equivalence. Use ``.is_equivalent()`` for actual check!
+
+        Parameters
+        ----------
+        other : `~astropy.cosmology.FLRW` subclass instance
+            The object in which to compare.
+
+        Returns
+        -------
+        `True` or `NotImplemented`
+            `True` if 'other' is of the same class and has matching parameters
+            and parameter values. `NotImplemented` otherwise.
+
+        """
+        # Note! this method is superseded by FlatFLRWMixin.__equiv__ in
+        # subclassses of FlatFLRWMixin.
+
+        if other.__class__ is not self.__class__:
+            return NotImplemented
+
+        # check all parameters in other match those in 'self' and 'other' has
+        # no extra parameters (latter part should never happen b/c same class)
+        sps, ops = self.info("parameters"), other.info("parameters")
+        params_eq = (all((np.all(ops[k] == v) for k, v in sps.items()))
+                     and not (sps.keys() ^ ops.keys()))
```

Comment:
`and sps.keys() == ops.keys()`.

Note that unlike for `__eq__`, where an error automatically leads to `False`, it may be best here to do the check on key equality first, so that we know all `ops[k]` are going to work.

Reply:
Agreed. I've changed all the tests accordingly.